drivers/gpio/gpio_cc32xx.c

```diff
@@ -187,6 +187,12 @@ static int gpio_cc32xx_pin_interrupt_configure(struct device *port,
 				int_type = GPIO_LOW_LEVEL;
 			}
 		}
+		/*
+		 * disable interrupt prior to changing int type helps
+		 * prevent spurious interrupts observed when switching
+		 * to level-based
+		 */
+		MAP_GPIOIntDisable(port_base, (1 << pin));
```

Comment:
I propose to place the line at the top of the function, i.e. upon entry interrupts will be disabled. We can drop then the `else` statement.

Reply:
I made the change. Thanks!